mcp/mcp-server/src/main/java/software/amazon/smithy/java/mcp/server/HttpMcpProxy.java

```diff
@@ -0,0 +1,165 @@
+    private final String name;
+    private final Supplier<Map<String, String>> headersSupplier;
+    private final Duration timeout;
+
+    private HttpMcpProxy(Builder builder) {
+        this.transport = builder.transport != null ? builder.transport : new JavaHttpClientTransport();
+        this.endpoint = URI.create(builder.url);
+        this.name = builder.name != null ? builder.name : sanitizeName(endpoint.getHost());
+        this.headersSupplier = builder.headersSupplier != null ? builder.headersSupplier : Map::of;
+        this.timeout = builder.timeout != null ? builder.timeout : Duration.ofSeconds(60);
+    }
+
+    private static String sanitizeName(String host) {
+        if (host == null) {
+            return "http-proxy-mcp";
+        }
+        return host.replaceAll("[^a-zA-Z0-9-]", "-");
+    }
+
+    public static class Builder {
+        private String url;
+        private String name;
+        private Supplier<Map<String, String>> headersSupplier;
+        private ClientTransport<HttpRequest, HttpResponse> transport;
+        private Duration timeout;
+
+        public Builder url(String url) {
+            this.url = url;
+            return this;
+        }
+
+        public Builder name(String name) {
+            this.name = name;
+            return this;
+        }
+
+        public Builder headers(Supplier<Map<String, String>> headersSupplier) {
+            this.headersSupplier = headersSupplier;
+            return this;
+        }
+
+        public Builder headers(Map<String, String> headers) {
+            this.headersSupplier = () -> headers;
+            return this;
+        }
+
+        public Builder transport(ClientTransport<HttpRequest, HttpResponse> transport) {
+            this.transport = transport;
+            return this;
+        }
+
+        public Builder timeout(Duration timeout) {
+            this.timeout = timeout;
+            return this;
+        }
+
+        public HttpMcpProxy build() {
+            if (url == null || url.isEmpty()) {
+                throw new IllegalArgumentException("URL must be provided");
+            }
+            return new HttpMcpProxy(this);
+        }
+    }
+
+    public static Builder builder() {
+        return new Builder();
+    }
+
+    @Override
+    public CompletableFuture<JsonRpcResponse> rpc(JsonRpcRequest request) {
+        return CompletableFuture.supplyAsync(() -> {
+            try {
+                if (request == null) {
+                    throw new IllegalArgumentException("JsonRpcRequest cannot be null");
+                }
+
+                byte[] body = JSON_CODEC.serializeToString(request).getBytes(StandardCharsets.UTF_8);
+                LOG.trace("Sending HTTP request to {}", endpoint);
+
+                HttpRequest.Builder requestBuilder = HttpRequest.builder()
+                        .uri(endpoint)
+                        .method("POST")
+                        .withAddedHeader("Content-Type", "application/json")
```

Comment:
You need to send a Accept header, which sends both `application/json` and `text/event-stream`